### PR TITLE
Refactor UI selection

### LINKED ISSUE #515: UI Filtering: Selection
### User Story

As a user, I want to be able to filter data down to a more granular level to visually see citation data for a region and time frame.

### Description

When a user selects an input field, the user needs to see a different UI matching with the corresponding input type.

A user must see the appropriate header matching the data type that the user had selected; Neighborhood Council if they had picked a neighborhood council data type and Zip Code if they had picked a Zip Code. 

On this new UI, there should be two main dropdowns: Citation Type and Date Range.

**Citation Type Dropdown Options:**
- Total # Citations
- Citations by Population (Density)
- Violation Type
- Total $ Fine Amount
- Fine $ Amount by Population (Density)
- Day of the Week

**Date Range Dropdown Options**

- Past 1 Year
- Year to Date
- Past 3 Months
- Past 1 Month
- All Time

As part of this new UI, there should also be a From and To Calendar link options. 

Please leave an area below this the date range drop down and calendar from and to blank for future graphs. Graphs are out of scope for this ticket.

### Acceptance Criteria

1. If a user selects neighborhood council as the input field, then the user would see a new UI appear with Citation type, Date Dropdowns, Date Range, and a graphical display of data.
2. If a user selects neighborhood council as the input field, then the user would see Neighborhood Council as the region type in the header. If a user selects zip code as the input field, then the user would see Zip Code as the region type in the header.
3. If a user selects on the dropdown, then the user would see the full list of dropdown options for both citation types and date range.

### Design References

[Figma](https://www.figma.com/file/R3N7mvDtlFbFgQm4OXcMih/Lucky-Parking?type=design&node-id=731-243&mode=design&t=O6uPiQpCRAKlc3Yd-0)

### Technical References

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gibsonliketheguitar**
### Description

Replace Picklist for citation-date-preset and citation-data-selection component
Styling bump
refactor calendar component with outside click closing modal
refactor stepper container with React.Children

### Related Issues

Resolves #515 

### Testing

- Ran testing suites
```
 PASS  packages/ui/src/components/stepper/stepper.test.tsx
 PASS  packages/ui/src/components/pick-list/pick-list.test.tsx
 PASS  packages/ui/src/components/input/input.test.tsx
 PASS  packages/ui/src/components/modal/modal.test.tsx
 PASS  packages/website/src/features/instructions/how-it-works-button.test.tsx
 FAIL  packages/website/src/widgets/header/header.test.tsx
```

### Checklist

- [x] I have followed all conventions outlined in our documentation.
- [x] I have fully tested my changes and confirmed that all new and existing tests pass.
- [x] I have written meaningful commit messages for all changes.
- [x] I have linted and formatted my changes to follow the code style of this repository.
- [ ] I have updated our documentation, accordingly.
- [x] I have checked currently opened pull requests to ensure that there are no pending pull request for the same changes or issues.
- [x] I have confirmed that this pull request fully meets the acceptance criteria for all related issues listed above.

### CODE REVIEW

File: packages/website/src/shared/ui/date-input/date-input.tsx [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
+  const [multiRefs, addMultiRef] = useMultiRefs();
   const [value, setValue] = useState<string | null>();
   const [isCalendarVisible, setCalendarVisible] = useState(false);
 
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      const arr = multiRefs() as HTMLElement[];
+      if (!arr.length) return;
+
+      //check if click event is within any of our refs
```
Comment (glenflorendo) [inline packages/website/src/shared/ui/date-input/date-input.tsx:17]: I understand what we're trying to accomplish here which is good, but are there any other approaches we can do?

**Review by glenflorendo — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/stepper/stepper-container.tsx [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+  const items = Children.toArray(children) || [];
+  const len = items.length;
 
   return (
     <div className="ml-4 flex flex-col">
-      {children?.map((child: any, idx: number) => {
-        const isLast = idx === len - 1;
-        return createElement(child.type, { key: idx, ...child.props, isCurrent: currentStep === idx, isLast });
+      {items.map((child, idx) => {
+        if (!isReactElement(child)) return <></>;
```
[the remark this answers is not in the document]
Comment (gibsonliketheguitar) [reply packages/ui/src/components/stepper/stepper-container.tsx]: Originally it was type guarding the react child.type. Found a better method. with isValidElement

File: packages/website/src/shared/ui/date-input/date-input.tsx (hunk 1 above)
Comment (gibsonliketheguitar) [reply packages/website/src/shared/ui/date-input/date-input.tsx]: Good catch. I did a refactor and found a better solution. I remove the add multi Refs and just refactored picklist to render in the body of the calendar. That way we don't need to use multiple refs.

https://github.com/hackforla/lucky-parking/pull/560/commits/ef10b01f7b636a5cf46a68ee7e24139d73ee28bf

One thing I think we need to do soon, is create a custom class for our z-indexes

**Comment by glenflorendo** [comment]
@gibsonliketheguitar Please let me know when this is ready for another review. Thanks!

**Comment by gibsonliketheguitar** [comment]
@glenflorendo it's ready

**Comment by glenflorendo** [comment]
@gibsonliketheguitar When I select an option in the picklist, the value doesn't change.

https://652dc9ddb28c970addba3753--lucky-parking-staging.netlify.app/

Could you also `Resolve conversation` on any discussion that is completed?

**Comment by gibsonliketheguitar** [comment]
> @gibsonliketheguitar When I select an option in the picklist, the value doesn't change.

Should be fixed now. 

> Could you also `Resolve conversation` on any discussion that is completed?

Can do

**Review by glenflorendo — APPROVED** [review]
Great work! This looks really good.

**Pull request merged**